File: templates/java_rest_api_template/src/main/java/com/template/restapi/error/ErrorResponse.java
/**
 * #############################################################################
 * ### Error response body
 * ###
 * ### @file ErrorResponse.java
 * ### @author Sebastian Russo
 * ### @date 2026
 * #############################################################################
 *
 * Uniform JSON error shape returned by the GlobalExceptionHandler. Using a
 * single record keeps every error response consistent for API consumers.
 *
 * Example serialized body:
 *   { "status": 429, "error": "Too Many Requests",
 *     "detail": "Request rate limit exceeded. Please try again later." }
 */
package com.template.restapi.error;

public record ErrorResponse(int status, String error, String detail) {
}

File: templates/csharp_rest_api_template/src/RestApiTemplate/Errors/ErrorResponse.cs
namespace Template.RestApi.Errors;

public sealed record ErrorResponse(int Status, string Error, string Detail)
{
    public static ErrorResponse BadRequest(string detail) => new(400, "bad_request", detail);

    public static ErrorResponse NotFound(string detail) => new(404, "not_found", detail);

    public static ErrorResponse TooManyRequests(string detail) => new(429, "rate_limit_exceeded", detail);

    public static ErrorResponse InternalServerError() =>
        new(500, "internal_server_error", "An unexpected error occurred.");
}
